<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="name" default="" />
  <arg name="portname" default="/dev/parachute" />
  <!-- baudrate: 9600 19200 38400 57600 115200 230400 460800 500000 576000 921600-->
  <arg name="baudrate" default="115200" /> 
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <arg name="swap_garmins" default="$(optenv SWAP_GARMINS false)" />

  <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_bacaprotocol_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the nodelet -->
    <node pkg="nodelet" type="nodelet" name="parachute" args="$(arg nodelet) baca_protocol/BacaProtocol $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">


      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>

      <rosparam file="$(find mrs_serial)/config/mrs_serial.yaml" />

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="baudrate" type="int" value="$(arg baudrate)" />
      <param name="portname" value="$(arg portname)"/>
      <param name="use_timeout" value="true"/>

      <param name="swap_garmins" value="$(arg swap_garmins)"/>

      <!-- Publishers -->
      <remap from="~range" to="/$(arg UAV_NAME)/garmin/range" />
      <remap from="~range_up" to="/$(arg UAV_NAME)/garmin/range_up" />
      <remap from="~profiler" to="profiler" />
      <remap from="~baca_protocol_out" to="~received_message" />

        <!-- Subscribers -->
      <remap from="~baca_protocol_in" to="~send_message" />
      <remap from="~raw_in" to="~send_raw_message" />

    </node>

  </group>

</launch>
